full-stack-tests/backend/src/perftest/ChangesetReader.test.ts

```diff
@@ -0,0 +1,118 @@
+        <ECSchemaReference name="BisCore" version="01.00" alias="bis"/>
+        <ECEntityClass typeName="TestElement">
+            <BaseClass>bis:GraphicalElement2d</BaseClass>
+            <ECProperty propertyName="name" typeName="string"/>
+        </ECEntityClass>
+    </ECSchema>`;
+    await rwIModel.importSchemaStrings([schema]);
+    rwIModel.channels.addAllowedChannel(ChannelControl.sharedChannelName);
+
+    // Create drawing model and category
+    const codeProps = Code.createEmpty();
+    codeProps.value = "DrawingModel";
+    await rwIModel.locks.acquireLocks({ shared: IModel.dictionaryId });
+    const [, drawingModelId] = IModelTestUtils.createAndInsertDrawingPartitionAndModel(rwIModel, codeProps, true);
+    let drawingCategoryId = DrawingCategory.queryCategoryIdByName(rwIModel, IModel.dictionaryId, "MyDrawingCategory");
+    if (undefined === drawingCategoryId)
+      drawingCategoryId = DrawingCategory.insert(rwIModel, IModel.dictionaryId, "MyDrawingCategory", new SubCategoryAppearance());
+
+    rwIModel.saveChanges();
+    await rwIModel.pushChanges({ description: "Initial Test Data Setup", accessToken: adminToken });
+
+    // Create changesets with different number of inserts
+    const testCases = [
+      { testCaseNum: 1, numElements:     1000 },
+      { testCaseNum: 2, numElements:    10000 },
+      { testCaseNum: 3, numElements:   100000 },
+      { testCaseNum: 4, numElements:  1000000 },
+      { testCaseNum: 5, numElements: 10000000 },];
+
+    const elementPropsTemplate = {
+      classFullName: "TestDomain:TestElement",
+      model: drawingModelId,
+      category: drawingCategoryId,
+      code: Code.createEmpty(),
+    };
+
+    for (const testCase of testCases) {
+      await rwIModel.locks.acquireLocks({ shared: drawingModelId });
+      for (let i = 0; i < testCase.numElements; i++) {
+        const elementProps = {
+          ...elementPropsTemplate,
+          name: `Element_${testCase.numElements}_${i}`,
+        };
+        assert.isTrue(Id64.isValidId64(rwIModel.elements.insertElement(elementProps)), `Failed to insert element ${elementProps.name}`);
+      }
+      rwIModel.saveChanges();
+      await rwIModel.pushChanges({ description: `Changeset with ${testCase.numElements} inserts`, accessToken: adminToken });
+    }
+
+    const targetDir = path.join(KnownTestLocations.outputDir, rwIModelId, "changesets");
+    const changesets = await HubMock.downloadChangesets({ iModelId: rwIModelId, targetDir }); // Get all changesets
+
+    for (const testCase of testCases) {
+      // Open the changesets one by one and read the changes
+      const reader = SqliteChangesetReader.openFile({ fileName: changesets[testCase.testCaseNum].pathname, db: rwIModel, disableSchemaCheck: true });
+      const adaptor = new ChangesetECAdaptor(reader);
+      const unifier = new PartialECChangeUnifier(reader.db);
```

Comment:
You should split the performance test into two now, one for the in-memory cache and one for the sqlite cache

Reply:
will this test work for inMemory case at all?  10 million cases seem to be aggressive for in memory.

Reply:
That's a good point. How about cut off the test cases for in-memory so that there's still a few, but don't bother with the larger number of changes.

Reply:
I made this change